ansible_base/authentication/authenticator_plugins/local.py

```diff
@@ -37,7 +39,15 @@ def authenticate(self, request, username=None, password=None, **kwargs):
         # This auth class doesn't create any new local users, so we just need to make sure
         # it has an AuthenticatorUser associated with it.
         if user:
-            AuthenticatorUser.objects.get_or_create(uid=username, user=user, provider=self.database_instance)
```

Comment:
so I guess my reading of the test failure is that `user=user` is still needed to satisfy the non-null constraint of the field.

Once that's fixed, what you're doing looks good.

Reply:
@AlanCoding Switched to re-using the existing `get_or_create_authenticator_user` function that we are already using for LDAP authenticator, as per @john-westcott-iv 's suggestion.